tests/integration/devfile/cmd_devfile_watch_test.go

```diff
@@ -72,18 +77,59 @@ var _ = Describe("odo devfile watch command tests", func() {
 
 	Context("when executing watch without a valid devfile", func() {
 		It("should fail", func() {
-			// Devfile push requires experimental mode to be set
-			helper.CmdShouldPass("odo", "preference", "set", "Experimental", "true")
 			output := helper.CmdShouldFail("odo", "watch", "--devfile", "fake-devfile.yaml")
 			Expect(output).To(ContainSubstring("The current directory does not represent an odo component"))
 		})
 	})
 
 	Context("when executing odo watch with devfile flag without experimental mode", func() {
 		It("should fail", func() {
-			helper.CopyExample(filepath.Join("source", "devfiles", "nodejs"), context)
+			helper.CmdShouldPass("odo", "preference", "set", "Experimental", "false", "-f")
+			helper.CmdShouldPass("odo", "create", "nodejs", "--project", namespace, cmpName)
+			helper.CopyExample(filepath.Join("source", "devfiles", "nodejs", "project"), context)
+			helper.CopyExampleDevFile(filepath.Join("source", "devfiles", "nodejs", "devfile.yaml"), filepath.Join(context, "devfile.yaml"))
 			output := helper.CmdShouldFail("odo", "watch", "--devfile", filepath.Join(context, "devfile.yaml"))
 			Expect(output).To(ContainSubstring("Error: unknown flag: --devfile"))
 		})
 	})
+
+	Context("when executing odo watch after odo push", func() {
+		It("should listen for file changes", func() {
+			helper.CmdShouldPass("odo", "create", "nodejs", "--project", namespace, cmpName)
+
+			helper.CopyExample(filepath.Join("source", "devfiles", "nodejs", "project"), context)
+			helper.CopyExampleDevFile(filepath.Join("source", "devfiles", "nodejs", "devfile.yaml"), filepath.Join(context, "devfile.yaml"))
+
+			output := helper.CmdShouldPass("odo", "push", "--devfile", "devfile.yaml", "--project", namespace)
+			Expect(output).To(ContainSubstring("Changes successfully pushed to component"))
+
+			watchFlag := ""
+			odoV2Watch := utils.OdoV2Watch{
+				CmpName:            cmpName,
+				StringsToBeMatched: []string{"Executing devbuild command", "Executing devrun command"},
+			}
+			// odo watch and validate
+			utils.OdoWatch(utils.OdoV1Watch{}, odoV2Watch, namespace, context, watchFlag, oc, "kube")
```

Comment:
Runner should be `kubectl` and `oc` both. Please make the change throughout the test.